### PR TITLE
[Transaction] Support transactional Id expiration

### PR DESCRIPTION
**Pull request by Demogorgon314**
### Motivation
#896

### Modifications
1. Support transactional Id expiration
2. Refactor `TransactionStateManagerTest`
3. Remove `txnLogStoreAndTCImmigrationTest`, it will add back in other PR.
4. Use TransactionConfig `TransactionMetadataTopicName` instead of `Topic.TRANSACTION_STATE_TOPIC_NAME`.

### CODE REVIEW

**Review by BewareMyPower — APPROVED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -118,6 +118,9 @@ This section lists configurations about the transaction.
 | brokerId                     | The broker ID that is used to create the producer ID.  | 1       |
 | txnLogTopicNumPartitions     | the number of partitions for the transaction log topic. | 50      |
 | txnAbortTimedOutTransactionCleanupIntervalMs | The interval in milliseconds at which to rollback transactions that have timed out. | 10000 |
+| transactionalIdExpirationMs | The time in ms that the transaction coordinator will wait without receiving any transaction status updates for the current transaction before expiring its transactional id. | 604800 |
```
Comment (Huanli-Meng) [inline docs/configuration.md:121]:
```suggestion
| transactionalIdExpirationMs | The time (in ms) that the transaction coordinator waits without receiving any transaction status updates for the current transaction before expiring its transactional ID. | 604800 |
```

File: docs/configuration.md [hunk 2]
```diff
@@ -118,6 +118,9 @@ This section lists configurations about the transaction.
...
+| transactionsRemoveExpiredTransactionalIdCleanupIntervalMs | The interval in milliseconds at which to remove transactions that have expired. | 3600 |
```
Comment (Huanli-Meng) [inline docs/configuration.md:122]:
```suggestion
| transactionsRemoveExpiredTransactionalIdCleanupIntervalMs | The interval (in ms) at which to remove expired transactions. | 3600 |
```

File: docs/configuration.md [hunk 3]
```diff
@@ -118,6 +118,9 @@ This section lists configurations about the transaction.
...
+| enableTransactionalIdExpiration | Whether to enable transactional id expiration. | true |
```
Comment (Huanli-Meng) [inline docs/configuration.md:123]:
```suggestion
| enableTransactionalIdExpiration | Whether to enable transactional ID expiration. | true |
```

**Review by Huanli-Meng — COMMENTED** [review]
leave my minor comments. PTAL

**Comment by Demogorgon314** [comment]
All comments are addressed, PTAL. Thanks. @Huanli-Meng

**Review by Huanli-Meng — APPROVED** [review]

**Pull request merged**